packages/theme/src/cli/utilities/theme-environment/storefront-session.ts

```diff
@@ -10,7 +10,9 @@ export async function isStorefrontPasswordProtected(storeURL: string): Promise<b
     redirect: 'manual',
   })
 
-  return response.status === 302
+  if (response.status !== 302) return false
```

Comment:
PR - [Integration test in sfr](https://github.com/Shopify/storefront-renderer/pull/31439)

Code Ref - [How the redirect works in sfr](https://github.com/search?q=org%3AShopify+redirect_to_password_page%21&ref=opensearch&type=code)